lib/tasks/helpers/importmap_helper.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+require "importmap/packager"
+
+module ImportmapHelper
+
+  class Packager < Importmap::Packager
+    VerifyError = Class.new(StandardError)
+
+    # ugh, it defines this on the singleton class...
+    self.endpoint = URI("https://api.jspm.io/generate")
+
+    # Copied from https://github.com/rails/importmap-rails/pull/237
+    def verify(package, url)
+      ensure_vendor_directory_exists
+
+      if vendored_package_path(package).file?
+        verify_vendored_package(package, url)
+      end
+    end
+
+    def verify_vendored_package(package, url)
+      vendored_body = vendored_package_path(package).read.strip
+      remote_body = load_package_file(package, url).strip
+
+      return true if vendored_body == remote_body
+
+      raise VerifyError.new("Vendored #{package}#{extract_package_version_from(url)} does not match remote #{url}")
+    end
+
+    def load_package_file(package, url)
+      response = Net::HTTP.get_response(URI(url))
+
+      if response.code == "200"
+        format_vendored_package(package, url, response.body)
+      else
+        handle_failure_response(response)
+      end
+    end
+
+    def format_vendored_package(package, url, source)
+      formatted = +""
+      if Gem::Version.new(Importmap::VERSION) > Gem::Version.new("2.0.1")
+        formatted.concat "// #{package}#{extract_package_version_from(url)} downloaded from #{url}\n\n"
+      end
+      formatted.concat remove_sourcemap_comment_from(source).force_encoding("UTF-8")
+      formatted
+    end
+
+    def save_vendored_package(package, url, source)
+      File.open(vendored_package_path(package), "w+") do |vendored_package|
+        vendored_package.write source
+      end
+    end
+
+    public :vendored_package_path
+  end
+end
```

Comment:
trailing newline

Reply:
ugh. I removed the auto-newline to work one Prism and I have been forgetting to re-enable it. (Prism has some parse situations where the number of trailing newlines affects compatibility with the existing ruby parser.)